### PR TITLE
ST:  tests for env vars in containers

### PR DESCRIPTION
**Pull request by Frawless**
### Type of change

- Enhancement / new feature

### Description

This PR introduce new set of tests, which covers feature of specify environment variables inside component containers.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

**Comment by Frawless** [comment]
@strimzi-ci run tests test_only testcase=ConnectS2IST,HttpBridgeST#testCustomAndUpdatedValues,MirrorMakerST#testCustomAndUpdatedValues,ConnectST#testCustomAndUpdatedValues,KafkaST#testCustomAndUpdatedValues profile=regression

**Comment by Frawless** [comment]
@strimzi-ci run tests test_only testcase=ConnectS2IST,HttpBridgeST#testCustomAndUpdatedValues,MirrorMakerST#testCustomAndUpdatedValues,ConnectST#testCustomAndUpdatedValues,KafkaST#testCustomAndUpdatedValues profile=regression

**Comment by Frawless** [comment]
@strimzi-ci run tests test_only testcase=ConnectS2IST,HttpBridgeST#testCustomAndUpdatedValues,MirrorMakerST#testCustomAndUpdatedValues,ConnectST#testCustomAndUpdatedValues,KafkaST#testCustomAndUpdatedValues profile=regression

**Comment by Frawless** [comment]
Current failures are cause by issue in teardown. It seems that `oc cluster up` on top of openstack doesn't clear all resources. I will try to fix this in https://github.com/strimzi/strimzi-kafka-operator/pull/2026

File: systemtest/src/main/java/io/strimzi/systemtest/AbstractST.java [hunk 1]
```diff
@@ -214,7 +215,7 @@ void replaceMirrorMakerResource(String resourceName, Consumer<KafkaMirrorMaker>
         replaceCrdResource(KafkaMirrorMaker.class, KafkaMirrorMakerList.class, DoneableKafkaMirrorMaker.class, resourceName, editor);
     }
 
-    void replaceBridgeResource(String resourceName, Consumer<KafkaBridge> editor) {
+    protected void replaceBridgeResource(String resourceName, Consumer<KafkaBridge> editor) {
```
Comment (Seequick1) [inline systemtest/src/main/java/io/strimzi/systemtest/AbstractST.java]: Is this change needed ? :)

File: systemtest/src/test/java/io/strimzi/systemtest/MirrorMakerST.java [hunk 2]
```diff
@@ -352,6 +355,109 @@ void testWhiteList() throws Exception {
                 receiveMessages(messagesCount, kafkaClusterTargetName, false, topicNotInWhitelist, null, kafkaClientsPodName), is(0));
     }
 
+    @Test
+    void testCustomAndUpdatedValues() {
+        testMethodResources().kafkaEphemeral(CLUSTER_NAME, 1, 1)
+                .editSpec()
+                .withNewEntityOperator()
```
Comment (Seequick1) [inline systemtest/src/test/java/io/strimzi/systemtest/MirrorMakerST.java]: just add tab two `.wthNewEntityOperator()` and `.endEntityOperator`

Comment (Frawless) [reply systemtest/src/test/java/io/strimzi/systemtest/MirrorMakerST.java]: I will add as part of review-commit :)

File: systemtest/src/main/java/io/strimzi/systemtest/AbstractST.java (hunk 1 above)
Comment (Frawless) [reply systemtest/src/main/java/io/strimzi/systemtest/AbstractST.java]: Not sure for 100% not, but when I implemented it, there were problem that bridge tests are "in different package" so we can't use private methods there (from `AbstractST`). I will double check after other reviews and make changes if it's not needed.

File: systemtest/src/main/java/io/strimzi/systemtest/AbstractST.java [hunk 3]
```diff
@@ -1029,8 +1030,76 @@ void checkContainerConfiguration(String podNamePrefix, String containerName, Str
             pods.forEach(pod -> {
                 pod.getSpec().getContainers().stream().filter(c -> c.getName().equals(containerName))
                         .forEach(container -> {
+                            List list = container.getEnv().stream().map((Function<EnvVar, Object>) EnvVar::getName).collect(Collectors.toList());
+                            assertTrue(list.contains(configKey));
+
                             container.getEnv().stream().filter(envVar -> envVar.getName().equals(configKey))
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/AbstractST.java]: Do we really need to go through the env vars twice? First to collect all the names and then to check value? Wouldn't doing it once ith a filter be enough?

File: systemtest/src/main/java/io/strimzi/systemtest/AbstractST.java [hunk 4]
```diff
@@ -1029,8 +1030,76 @@ void checkContainerConfiguration(String podNamePrefix, String containerName, Str
...
-                                    .forEach(envVar -> assertEquals(config, envVar.getValue()));
+                                    .forEach(envVar -> {
+                                        LOGGER.info("Check specific configuration: {}", envVar.getName());
+                                        for (Map.Entry<String, Object> entry : config.entrySet()) {
+                                            assertThat(envVar.getValue().contains(entry.getKey() + "=" + entry.getValue() + "\n"), is(true));
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/AbstractST.java]: So, is this method used only to check that env var with the user suplied properties for the Kafka / Zoo properties file? You should make it more clear. Because up to until this line I expected that container configuration means the fileds of the container and not the KAfka/Zoo configuration.

File: systemtest/src/main/java/io/strimzi/systemtest/AbstractST.java [hunk 5]
```diff
@@ -1029,8 +1030,76 @@ void checkContainerConfiguration(String podNamePrefix, String containerName, Str
...
+        } else {
+            fail("Pod with prefix " + podNamePrefix + " in name, not found");
+        }
+    }
+
+    /**
+     * Verifies container configuration by environment key
+     * @param podNamePrefix Name of pod where container is located
+     * @param containerName The container where verifying is expected
+     * @param config Expected configuration
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/AbstractST.java]: Again, can wwe be more specific what exqctly is the expected config? Is that a map with environment variables here? Or something else?

File: systemtest/src/main/java/io/strimzi/systemtest/AbstractST.java [hunk 6]
```diff
@@ -1029,8 +1030,76 @@ void checkContainerConfiguration(String podNamePrefix, String containerName, Str
...
+            fail("Pod with prefix " + podNamePrefix + " in name, not found");
+        }
+    }
+
+    /**
+     * Verifies readinessProbe and livenessProbe properties in expected container
+     * @param podNamePrefix Prefix of pod name where container is located
+     * @param containerName The container where verifying is expected
+     * @param initialDelaySeconds expected value for property initialDelaySeconds
+     * @param timeoutSeconds expected value for property timeoutSeconds
```
Comment (scholzj) [inline systemtest/src/main/java/io/strimzi/systemtest/AbstractST.java]: I guess you should update the JavaDoc.

File: systemtest/src/main/java/io/strimzi/systemtest/AbstractST.java (hunk 4 above)
Comment (Frawless) [reply systemtest/src/main/java/io/strimzi/systemtest/AbstractST.java]: it checks env variable from containers, which is passed into configuration files for bridge/kafka/zoo/mm. So maybe rename it to `checkComponentConfiguration()` and second method to `checkSpecificVariablesInContainer()` ?

Comment (scholzj) [reply systemtest/src/main/java/io/strimzi/systemtest/AbstractST.java]: Yeah ... I think renaming it would help.
